llvm/include/llvm/IR/Attributes.td

```diff
@@ -22,290 +22,323 @@ def ParamAttr : AttrProperty;
 /// Can be used as return attribute.
 def RetAttr : AttrProperty;
 
+
+
+/// Intersection rules. Used for example in sinking/hoisting two
+/// callbases to find a set of attributes that apply to both.
+/// Note, there are some attributes we can (probably) legally drop
+/// but are intentionally excluded as of now. Those include:
+///     - initializes
+///     - allockind
+///     - allocsize
+///     - minsize
+///     - optsize
+///     - optnone
+///     - optdebug
+///     - optforfuzzing
```

Comment:
We shouldn't list things here. Instead, if the used intersection rule for an attribute is non-obvious, add a comment next to the definition to explain it.